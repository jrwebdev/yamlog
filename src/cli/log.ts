import prompt from '../util/prompt';
import log from '../lib/log';

log('breaking', 'test change 123');

const run = async () => {
  const answers = await prompt();
  log(answers.type, answers.details);
  if (answers.logAnother) {
    run();
  }
};

run();
