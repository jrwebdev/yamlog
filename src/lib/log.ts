import { ChangelogConfig } from '../types/config';
import { Change, ChangeType } from '../types/changelog';
import * as changelog from '../util/changelog';

export default (
  changeType: ChangeType,
  change: Change,
  config: ChangelogConfig = {}
) =>
  changelog.addChange(changeType, change, {
    unreleasedDir: config.unreleasedDir || '',
  });
