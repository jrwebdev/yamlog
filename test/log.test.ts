import { createMockProject } from './test-utils';

import log from '../src/lib/log';

let mockProject: any;

beforeEach(() => {
  mockProject = createMockProject();
  expect.hasAssertions();
});

afterEach(async () => {
  const changelog = await mockProject.readChangelog();
  mockProject.teardown();
  expect(changelog).toMatchSnapshot();
});

it('should create a new changelog.yaml file and add a change', async () => {
  mockProject.setup();
  log('fix', 'Fix 1');
});

it('should create a new unreleased section and add a change', async () => {
  mockProject.setup({
    '1.0.0': {
      metadata: { date: '2017-01-01' },
      feature: ['Feature 1'],
    },
  });
  // TODO: Fix failing test
  log('fix', 'Fix 1');
});

it('should add a change to an existing unreleased section', async () => {
  mockProject.setup({
    unreleased: {
      feature: ['Feature 1'],
    },
    '1.0.0': {
      metadata: { date: '2017-01-01' },
      feature: ['Feature 1'],
    },
  });
  // TODO: Fix failing test
  log('fix', 'Fix 1');
});

it('should add a fix', async () => {
  mockProject.setup({
    unreleased: {
      fix: ['Fix 1'],
    },
    '1.0.0': {
      metadata: { date: '2017-01-01' },
      feature: ['Feature 1'],
    },
  });
  // TODO: Fix failing test
  log('fix', 'Fix 2');
});

it('should add a feature', async () => {
  mockProject.setup({
    unreleased: {
      fix: ['Fix 1'],
    },
    '1.0.0': {
      metadata: { date: '2017-01-01' },
      feature: ['Feature 1'],
    },
  });
  // TODO: Fix failing test
  log('feature', 'Feature 1');
});

it('should add a breaking change', async () => {
  mockProject.setup({
    unreleased: {
      feature: ['Feature 1'],
      fix: ['Fix 1'],
    },
    '1.0.0': {
      metadata: { date: '2017-01-01' },
      feature: ['Feature 1'],
    },
  });
  // TODO: Fix failing test
  log('breaking', 'Breaking Change 1');
});
