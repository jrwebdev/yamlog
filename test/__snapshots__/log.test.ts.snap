// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add a breaking change 1`] = `
unreleased:
  feature:
    - Feature 1
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should add a change to an existing unreleased section 1`] = `
unreleased:
  feature:
    - Feature 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should add a feature 1`] = `
unreleased:
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should add a fix 1`] = `
unreleased:
  fix:
    - Fix 1
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;

exports[`should create a new changelog.yaml file and add a change 1`] = `
unreleased:
  fix:
    - Fix 1

`;

exports[`should create a new unreleased section and add a change 1`] = `
1.0.0:
  metadata:
    date: '2017-01-01'
  feature:
    - Feature 1

`;
